packages/hardhat-core/src/builtin-tasks/node.ts

```diff
@@ -79,10 +164,32 @@ task(TASK_NODE, "Starts a JSON-RPC server on top of Hardhat Network")
     8545,
     types.int
   )
+  .addOptionalParam(
+    "forkUrl",
+    "The URL of the JSON-RPC server to fork from",
+    undefined,
+    types.string
+  )
+  .addOptionalParam(
+    "forkBlockNumber",
+    "The block number to fork from",
+    undefined,
+    types.int
+  )
   .setAction(
     async (
-      { hostname, port },
-      { artifacts, network, hardhatArguments, config }
+      {
+        forkBlockNumber,
+        forkUrl,
+        hostname,
+        port,
+      }: {
+        forkBlockNumber?: number;
+        forkUrl?: string;
+        hostname: string;
+        port: number;
+      },
+      { config, hardhatArguments, network, run }
     ) => {
       if (
         network.name !== HARDHAT_NETWORK_NAME &&
```

Comment:
This expression can now be simplified to `network.name !== HARDHAT_NETWORK_NAME && hardhatArguments.network !== undefined`. We don't do anymore what the comment describes. If the user doesn't use `--network`, ` hardhatArguments.network` is `undefined`.

Reply:
Done in e614962c, you might want to check the updated comment to see if it makes sense.

Reply:
Looks good now